modulemd/include/modulemd-1.0/private/modulemd-yaml.h

```diff
@@ -75,46 +75,50 @@ typedef gboolean (*ModulemdParsingFunc) (yaml_parser_t *parser,
-      g_debug (msg);                                                          \
+      g_debug (__VA_ARGS__);                                                  \
       result = FALSE;                                                         \
       goto error;                                                             \
     }                                                                         \
   while (0)
 
-#define MMD_ERROR_RETURN_FULL(_error, type, msg)                              \
+#define MMD_ERROR_RETURN_FULL(_error, type, ...)                              \
   do                                                                          \
     {                                                                         \
-      g_debug (msg);                                                          \
-      g_set_error_literal (_error, MODULEMD_YAML_ERROR, type, msg);           \
+      g_debug (__VA_ARGS__);                                                  \
+      g_set_error (_error, MODULEMD_YAML_ERROR, type, __VA_ARGS__);           \
       result = FALSE;                                                         \
       goto error;                                                             \
     }                                                                         \
   while (0)
 
-#define MMD_YAML_ERROR_RETURN(_error, msg)                                    \
+#define MMD_YAML_ERROR_RETURN(_error, ...)                                    \
   do                                                                          \
     {                                                                         \
-      g_debug (msg);                                                          \
-      g_set_error_literal (                                                   \
-        _error, MODULEMD_YAML_ERROR, MODULEMD_YAML_ERROR_PARSE, msg);         \
+      g_debug (__VA_ARGS__);                                                  \
+      g_set_error (                                                           \
+        _error, MODULEMD_YAML_ERROR, MODULEMD_YAML_ERROR_PARSE, __VA_ARGS__); \
       result = FALSE;                                                         \
       goto error;                                                             \
     }                                                                         \
   while (0)
 
-#define YAML_EMITTER_EMIT_WITH_ERROR_RETURN(emitter, event, _error, msg)      \
+#define YAML_EMITTER_EMIT_WITH_ERROR_RETURN(emitter, event, _error, ...)      \
   do                                                                          \
     {                                                                         \
       if (!yaml_emitter_emit (emitter, event))                                \
         {                                                                     \
+          gchar *formatted = g_strdup_printf (__VA_ARGS__);                   \
```

Comment:
Please use `g_autofree gchar *__formatted = g_strdup_printf (__VA_ARGS__);` here. I'm trying to move towards using the GLib auto pointers wherever possible, that way we don't end up leaking if we add an error return somewhere.

Reply:
The added commit 8cdf3ad2e27145079c07d61d0c3ca1b382975857 uses `g_autofree` rather than freeing the pointers manually.

Reply:
And commit 644b944 should get the formatting right. :wink: